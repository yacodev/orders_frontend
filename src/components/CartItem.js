import styled from "@emotion/styled";
import { Image } from "./UI/Image";
import { SemiBoldS, SemiBoldM } from "./UI/Typography";
import { Counter } from "./UI/Counter";
import { useState, useEffect } from "react";
import { ADD_ITEM } from "../constants";
import capitalize from "../utils/capitalizeString";

const CardContainer = styled.div`
  display: flex;
  gap: 20px;
  width: 315px;
  height: 102px;
  padding: 20px;
  background-color: white;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.03);
  border-radius: 20px;
`;

const FlexContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: ${(props) => props.direction || "row"};
  justify-content: ${(props) => props.justify || "center"};
  gap: 8px;
`;

export default function CartItem({
  id,
  imgURL,
  name,
  price,
  updateOrder,
  removeFromCart,
}) {
  const [count, setCount] = useState(1);
  const subTotal = (price * count) / 100;

  useEffect(() => {
    if (count === 0) removeFromCart(id);
  }, [count, id, removeFromCart]);

  function updateCount(type) {
    if (type === ADD_ITEM) {
      setCount(count + 1);
      updateOrder.addProduct(price / 100, id);
    } else {
      setCount(count - 1);
      updateOrder.substractProduct(price / 100, id);
    }
  }

  return (
    <CardContainer>
      <Image src={imgURL} alt="" size="sm" />
      <FlexContainer direction="column">
        <SemiBoldS>{capitalize(name)}</SemiBoldS>
        <FlexContainer justify="space-between">
          <SemiBoldM color="#FA4A0C">${subTotal}</SemiBoldM>
          <Counter count={count} updateCount={updateCount} />
        </FlexContainer>
      </FlexContainer>
    </CardContainer>
  );
}
