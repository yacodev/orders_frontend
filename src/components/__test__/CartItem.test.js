import renderer from "react-test-renderer";
import CartItem from "../CartItem";

describe("CartItem", () => {
  it("renders correctly CartItem", () => {
    const image =
      "https://img.freepik.com/free-photo/top-view-green-cream-soups_23-2148519096.jpg";
    const name = "green cream";
    const price = "4513";
    const tree = renderer
      .create(<CartItem imgURL={image} name={name} price={price} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});