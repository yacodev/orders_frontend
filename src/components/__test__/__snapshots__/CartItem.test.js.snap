// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem renders correctly CartItem 1`] = `
<div
  className="css-1e0t6nl"
>
  <img
    alt=""
    className="css-bbanfq"
    size="sm"
    src="https://img.freepik.com/free-photo/top-view-green-cream-soups_23-2148519096.jpg"
  />
  <div
    className="css-9o085f"
    direction="column"
  >
    <p
      className="css-12jf3i4"
    >
      Green cream
    </p>
    <div
      className="css-u92498"
    >
      <p
        className="css-1na6q0d"
        color="#FA4A0C"
      >
        $
        45.13
      </p>
      <div
        className="css-okb5hw"
      >
        <button
          className="css-8th4lx"
          onClick={[Function]}
        >
           - 
        </button>
        <p
          className="css-1u7p27y"
        >
           
          1
           
        </p>
        <button
          className="css-8th4lx"
          onClick={[Function]}
        >
           + 
        </button>
      </div>
    </div>
  </div>
</div>
`;
